### PR TITLE
Revert "PERF: nancorr pearson (#42761)"

### LINKED ISSUE #45640: BUG: REGRESSION: DataFrame.corr() floating point inaccuracy
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

rng = np.random.default_rng(1)
df = pd.DataFrame(rng.normal(size=(100, 5)))

print(1 - np.diag(df.corr().abs()).min())
# pandas 1.4.0
# -4.440892098500626e-16
# pandas 1.3.5
# 0.0
```

### Issue Description

With pandas 1.4.0,  `df.corr()` returns a matrix where the diagonal is not exactly 1 down to floating point precision.

In pandas 1.3.5 the diagonal of `df.corr()` was exactly 1.

The example above show the difference.

This causes issues when using the `dist = 1 - df.corr().abs()` as a distance matrix for clustering. In particular the call to `scipy.spatial.distance.squareform(dist)` raises an error with pandas 1.4.0 when the `dist` diagonal is not exactly 0.

### Expected Behavior

The diagonal of `df.corr()` should be exactly 1 down to floating point accuracy

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #45640
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.4.1.rst [hunk 1]
```diff
@@ -32,7 +32,7 @@ Bug fixes
 
 Other
 ~~~~~
--
+- Reverted performance speedup of nancorr improvement (:issue:`45640`, :issue:`42761`)
```
Comment (mzeitlin11) [inline doc/source/whatsnew/v1.4.1.rst:35]: Could add a bit more specificity, maybe
```suggestion
- Reverted performance speedup of :meth:`DataFrame.corr` for ``method=pearson`` to fix precision regression (:issue:`45640`, :issue:`42761`)
```

**Review by mzeitlin11 — APPROVED** [review]
LGTM, thanks for fixing @phofl!

File: doc/source/whatsnew/v1.4.1.rst (hunk 1 above)
Comment (phofl) [reply doc/source/whatsnew/v1.4.1.rst]: thx

**Comment by jreback** [comment]
thanks @phofl 

I guess we should re-open the *original* issue though that was a bit off point.

**Comment by phofl** [comment]
I think it was not closer back then, just referenced

**Pull request merged**